engine/graphics/src/opengl/graphics_opengl.cpp

```diff
@@ -2856,84 +2867,45 @@ static void LogFrameBufferError(GLenum status)
         return HANDLE_RESULT_OK;
     }
 
-    static void OpenGLSetTexture(HTexture texture, const TextureParams& params)
+    static inline void GetOpenGLSetTextureParams(OpenGLContext* context, TextureFormat format, GLint& gl_internal_format, GLenum& gl_format, GLenum& gl_type)
     {
-        DM_PROFILE(__FUNCTION__);
-
-        // validate write accessibility for format. Some format are not garuanteed to be writeable
-        switch (params.m_Format)
-        {
-            case TEXTURE_FORMAT_DEPTH:
-                dmLogError("TEXTURE_FORMAT_DEPTH is not a valid argument for SetTexture");
-                return;
-            case TEXTURE_FORMAT_STENCIL:
-                dmLogError("TEXTURE_FORMAT_STENCIL is not a valid argument for SetTexture");
-                return;
-            default:
-                break;
-        }
-
-        // Responsibility is on caller to not send in too big textures.
-        assert(params.m_Width <= g_Context->m_MaxTextureSize);
-        assert(params.m_Height <= g_Context->m_MaxTextureSize);
-
-        int unpackAlignment = 4;
-        /*
-         * For RGB-textures the row-alignment may not be a multiple of 4.
-         * OpenGL doesn't like this by default
-         */
-        if (params.m_Format != TEXTURE_FORMAT_RGBA && !IsTextureFormatCompressed(params.m_Format))
-        {
-            uint32_t bytes_per_row = params.m_Width * dmMath::Max(1U, GetTextureFormatBitsPerPixel(params.m_Format)/8);
-            if (bytes_per_row % 4 == 0) {
-                // Ok
-            } else if (bytes_per_row % 2 == 0) {
-                unpackAlignment = 2;
-            } else {
-                unpackAlignment = 1;
+    #ifdef __EMSCRIPTEN__
+        #define WEBGL1_WORKAROUND(fmt) \
+            if (!context->m_IsGles3Version) \
+            { \
+                gl_internal_format = fmt; \
```

Comment:
For backwards compatability we need to do a workaround for WebGL1 here to use the "old" formats. We can't rely on compile time constants since they differ between the context versions..